Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/BaseCollection.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Collections
+{
+    public abstract class BaseCollection : MonoBehaviour
+    {
+        #region public members
+        /// <summary>
+        /// Action called when collection is updated
+        /// </summary>
+        public Action<BaseCollection> OnCollectionUpdated;
```

Comment:
Should we update this to be an event routed through the MRTK?

Reply:
I'm not what we'd gain that'd be worth the perf impact of routing local events like this through the event system. I'm sure @radicalad knows more about the use of this, it seems to me like mostly the objects in the collection or related objects would care about this event, instead of routing it to every global/modal/etc listener as well.

Reply:
I was mostly just curious. I don't mind keeping the action, but also adding an event.

Reply:
I'm kinda into it purely for an excuse to get to know the new system better. As far as use case, I get why its there in theory, but I've never used it myself, I almost scrapped it when I revisited it for this PR.